SecretManagement.LastPass.Extension/SecretManagement.LastPass.Extension.psm1

```diff
@@ -79,46 +116,66 @@ function Set-Secret
         [Parameter(ValueFromPipelineByPropertyName)]
         [hashtable] $AdditionalParameters
     )
-    if($Secret -is [string]) {
-        $Secret = @{
-            URL = "http://sn"
-            Notes = $Secret
-        }
-    }
-
     $sb = [System.Text.StringBuilder]::new()
-    if($Secret.UserName) {
-        $sb.Append("Username: ").AppendLine($Secret.UserName)
-        $sb.Append("login: ").AppendLine($Secret.UserName)
+    
+    
+    if ($Secret -is [string]) {
+        $Secret = @{Notes = $Secret}
+    } elseif ($Secret -is [pscredential]) {
+        $Secret = @{Username = $Secret.Username; Password = $Secret.GetNetworkCredential().password}
     }
 
-    if($Secret.Password) {
```

Comment:
Everything is a hashtable  (mostly) but in the case it is  a String of PSCredential, they are converted so the block below is not needed anymore.